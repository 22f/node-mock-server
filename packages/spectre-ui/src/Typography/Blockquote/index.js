// @flow
/* global props defaultProps */

import "spectre.css";
import React from "react";

import type { BlockquotePropsType } from "../../../spectre-ui.js.flow";

const Blockquote = ({ children, className }: BlockquotePropsType): React$Element<*> => (
	<blockquote className={className}>{children}</blockquote>
);

Blockquote.displayName = "Typography/Blockquote";

export default Blockquote;
