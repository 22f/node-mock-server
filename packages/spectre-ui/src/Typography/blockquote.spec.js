/* global describe it expect */

import React from "react";
import { shallow } from "enzyme";
import Blockquote from "./blockquote";

const props: Object = {
	children: <div />
};

describe("Blockquote", () => {
	describe("render", () => {
		it("snapshot: default", () => {
			const _props = { ...props };
			const tree = shallow(<Blockquote {..._props} />);
			expect(tree).toMatchSnapshot();
		});
		it("snapshot: no children", () => {
			const _props = { ...props, children: undefined };
			const tree = shallow(<Blockquote {..._props} />);
			expect(tree).toMatchSnapshot();
		});
	});
});
