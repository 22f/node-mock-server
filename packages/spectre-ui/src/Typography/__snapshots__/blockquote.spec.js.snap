// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote render snapshot: default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <blockquote
    className=""
>
    <div />
</blockquote>,
  "nodes": Array [
    <blockquote
      className=""
>
      <div />
</blockquote>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Blockquote>
        <div />
</Blockquote>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Blockquote {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div />,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <blockquote
          className=""
>
          <div />
</blockquote>,
        "_debugID": 2,
        "_renderedOutput": <blockquote
          className=""
>
          <div />
</blockquote>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Blockquote>
    <div />
</Blockquote>,
}
`;

exports[`Blockquote render snapshot: no children 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <blockquote
    className=""
/>,
  "nodes": Array [
    <blockquote
      className=""
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Blockquote />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Blockquote {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": undefined,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <blockquote
          className=""
/>,
        "_debugID": 4,
        "_renderedOutput": <blockquote
          className=""
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Blockquote />,
}
`;
